docs/guides/server/hostname.adoc

```diff
@@ -3,141 +3,127 @@
-<@kc.start parameters="--hostname=mykeycloak --http-enabled=true --proxy-headers=forwarded|xforwarded"/>
+* Frontend
+* Backend
+* Administration
 
-=== Exposing the server without a proxy
+If you want to work with either of these endpoints, you need to set the base URL. The base URL consists of a several parts:
 
-In this example, the server is running without a proxy and exposed using a URL using HTTPS.
+* a scheme (e.g. https protocol)
+* a hostname (e.g. example.keycloak.org)
+* a port (e.g. 8443)
+* a path (e.g. /auth)
 
-.{project_name} configuration:
-<@kc.start parameters="--hostname-url=https://mykeycloak"/>
+The base URL for each group has an important impact on how tokens are issued and validated, on how links are created for actions that require the user to be redirected to {project_name} (for example, when resetting password through email links), and, most importantly, how applications will discover these endpoints when fetching the OpenID Connect Discovery Document from `realms/{realm-name}/.well-known/openid-configuration`.
 
-It is highly recommended using a TLS termination proxy in front of the server for security and availability reasons. For more details,
-see <@links.server id="reverseproxy"/>.
+=== Frontend
 
-=== Forcing backend endpoints to use the same URL the server is exposed
+Users and applications use the frontend URL to access {project_name} through a front channel. The front channel is a publicly accessible communication channel. For example browser-based flows (accessing the login page, clicking on the link to reset a password or binding the tokens).
 
-In this example, backend endpoints are exposed using the same URL used by the server so that clients always fetch the same URL
-regardless of the origin of the request.
+=== Backend
 
-.{project_name} configuration:
-<@kc.start parameters="--hostname=mykeycloak --hostname-strict-backchannel=true"/>
+The backend endpoints are those accessible through a public domain or through a private network. They're related to direct backend communication between {project_name} and a client (an application secured by {project_name}). Such communication might be over a local network, avoiding a reverse proxy. Examples of the endpoints that belong to this group are the authorization endpoint, token and token introspection endpoint, userinfo endpoint, JWKS URI endpoint, etc.
 
-=== Exposing the server using a port other than the default ports
 
-In this example, the server is accessible using a port other than the default ports.
+=== Administration
 
-.{project_name} configuration:
-<@kc.start parameters="--hostname-url=https://mykeycloak:8989"/>
+Similarly to the base frontend URL, you can also set the base URL for resources and endpoints available through the management interface such as the administration console. The server exposes the administration console and static resources using a specific URL. This URL is used for redirect URLs, loading resources (CSS, JS), etc.
```

Comment:
@Pepo48 What is meant here by '_available through the management interface such as the administration console_'? I'd like to prevent any name collisions with the management interface we have already specified[1]. It's not even clear to me what management interface it is related to. Do you mean the external REST API for the admin console? 

I'd remove the 'management interface' to prevent confusion. @Pepo48 WDYT?



[1] https://github.com/keycloak/keycloak/blob/main/docs/guides/server/management-interface.adoc

Reply:
@mabartos addressed.